Library/Formula/gdb.rb

```diff
@@ -25,12 +25,13 @@ class Gdb < Formula
   end
 
   option "with-brewed-python", "Use the Homebrew version of Python"
+  option "with-brewed-readline", "Use the Homebrew version of Readline"
   option "with-version-suffix", "Add a version suffix to program"
   option "with-all-targets", "Build with support for all targets"
 
   depends_on "pkg-config" => :build
-  depends_on "readline"
-  depends_on "xz"
+  depends_on "python" if build.with? "brewed-python"
+  depends_on "readline" if build.with? "brewed-readline"
```

Comment:
I had a similar though earlier but was not sure on how to word it in options not just the description. The formula uses "--with-python=/usr" by default, and using a similar option, `--with-python`, to install a `Homebrewed` python might confuse the end user. I can reword the options, if it is unacceptable as per `Homebrew` conventions.

By the way, I had tried to link `liblzma` using `--with-liblzma-dir` but it just doesn't seem to work. I'll check on it later.